src/plugins/cross-references.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * © 2024 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Plugin } from 'prosemirror-state';
+import { Decoration, DecorationSet } from 'prosemirror-view';
+import { schema } from '@manuscripts/transform';
+
+export default () => {
+  return new Plugin({
+    state: {
+      init(_, state) {
+        // Initialize decorations with an empty set
+        return DecorationSet.empty;
+      },
+      apply(tr, oldState) {
+        let decoSet = oldState;
+
+        if (tr.docChanged) {
```

Comment:
we can check if state of the object plugin changed from new state and do an update just for the changed node. like this: 
```
import { isEqual } from 'lodash'
    
const oldTargets = objectsKey.getState(oldState)
const newTargets = objectsKey.getState(newState)

if (isEqual(oldTargets, newTargets)) {
```

Reply:
OK... it is updated now

Reply:
just replace or with and `if (tr.docChanged && !isEqual(oldTargets, newTargets))`